crates/bevy_animation/src/animatable.rs

```diff
@@ -0,0 +1,211 @@
+        for input in inputs {
+            if input.additive {
+                value += input.weight * Vec3A::from(input.value);
+            } else {
+                value = Vec3A::interpolate(&value, &Vec3A::from(input.value), input.weight);
+            }
+        }
+        Self::from(value)
+    }
+}
+
+impl Animatable for bool {
+    #[inline]
+    fn interpolate(a: &Self, b: &Self, t: f32) -> Self {
+        util::step_unclamped(*a, *b, t)
+    }
+
+    #[inline]
+    fn blend(inputs: impl Iterator<Item = BlendInput<Self>>) -> Self {
+        inputs
+            .max_by(|a, b| FloatOrd(a.weight).cmp(&FloatOrd(b.weight)))
+            .map(|input| input.value)
+            .unwrap_or(false)
+    }
+}
+
+impl Animatable for HandleId {
+    #[inline]
+    fn interpolate(a: &Self, b: &Self, t: f32) -> Self {
+        util::step_unclamped(*a, *b, t)
+    }
+
+    #[inline]
+    fn blend(inputs: impl Iterator<Item = BlendInput<Self>>) -> Self {
+        inputs
+            .max_by(|a, b| FloatOrd(a.weight).cmp(&FloatOrd(b.weight)))
+            .map(|input| input.value)
+            .expect("Attempted to blend HandleId with zero inputs.")
+    }
+}
+
+impl<T: Asset> Animatable for Handle<T> {
+    #[inline]
+    fn interpolate(a: &Self, b: &Self, t: f32) -> Self {
+        util::step_unclamped(a.clone_weak(), b.clone_weak(), t)
+    }
+
+    #[inline]
+    fn blend(inputs: impl Iterator<Item = BlendInput<Self>>) -> Self {
+        inputs
+            .max_by(|a, b| FloatOrd(a.weight).cmp(&FloatOrd(b.weight)))
+            .map(|input| input.value)
+            .expect("Attempted to blend Handle with zero inputs.")
+    }
+
+    fn post_process(&mut self, world: &World) {
+        // Upgrade weak handles into strong ones.
+        if self.is_strong() {
+            return;
+        }
+        *self = world
+            .get_resource::<Assets<T>>()
+            .expect(
+                "Attempted to animate a Handle<T> without the corresponding Assets<T> resource.",
+            )
+            .get_handle(self.id());
+    }
+}
+
+impl Animatable for Transform {
+    fn interpolate(a: &Self, b: &Self, t: f32) -> Self {
+        Self {
+            translation: Vec3::interpolate(&a.translation, &b.translation, t),
+            rotation: Quat::interpolate(&a.rotation, &b.rotation, t),
+            scale: Vec3::interpolate(&a.scale, &b.scale, t),
+        }
+    }
+
+    fn blend(inputs: impl Iterator<Item = BlendInput<Self>>) -> Self {
+        let mut translation = Vec3A::ZERO;
+        let mut scale = Vec3A::ZERO;
+        let mut rotation = Quat::IDENTITY;
+
+        for input in inputs {
+            if input.additive {
+                translation += input.weight * Vec3A::from(input.value.translation);
+                scale += input.weight * Vec3A::from(input.value.scale);
+                rotation = (input.value.rotation * input.weight) * rotation;
+            } else {
+                translation = Vec3A::interpolate(
```

Comment:
Vec3A vs Vec3 is a difference in just padding and alignment. AFAICT, reading a Vec3 as if it was a Vec3A is primarily a unaligned read and a mask, which should be pretty cheap compared to doing the same interpolation operations on each component. As @alice-i-cecile commented, we should probably benchmark this.

Reply:
This requires a load from CPU register to SIMD one, then a SIMD operation, then a store back. This extra load/store is almost surely faster if you do multiple operations while the values are in the SIMD registers, but for a single one it's unclear whether it's worth it (and probably depends on platform). In general when working with SIMD you should maximize grouped operations while values are in registers, and try to load only at start, store at end. So overhaul we should try to entirely avoid `Vec3` inside the animation pipeline and use ,`Vec3A` everywhere, so we keep the values loaded in registers. We should even consider interpolating first all translations, then all rotations, etc. so that we can batch the calculations.